include/verilated_types.h

```diff
@@ -1511,6 +1550,154 @@ std::string VL_TO_STRING(const VlUnpacked<T_Value, T_Depth>& obj) {
     return obj.to_string();
 }
 
+//===================================================================
+// Helper to apply the given indices to a target expression
+
+template <size_t Curr, size_t Rank, typename T_Target>
+struct VlApplyIndices final {
+    VL_ATTR_ALWINLINE
+    static auto& apply(T_Target& target, const size_t* indicesp) {
+        return VlApplyIndices<Curr + 1, Rank, decltype(target[indicesp[Curr]])>::apply(
+            target[indicesp[Curr]], indicesp);
+    }
+};
+
+template <size_t Rank, typename T_Target>
+struct VlApplyIndices<Rank, Rank, T_Target> final {
+    VL_ATTR_ALWINLINE
+    static T_Target& apply(T_Target& target, const size_t*) { return target; }
+};
+
+//===================================================================
+// Commit queue for NBAs - currently only for unpacked arrays
+
+template <class T_Target, bool Partial>
+class VlNBACommitQueue;
+
+// Whole element updates only
+template <class T_Target>
+class VlNBACommitQueue<T_Target, /* Partial: */ false> final {
```

Comment:
Please add standard // MEMBERS, // CONSTRUCTORS, // METHODS comments.

Need more comments on what this is doing, if it took me 10 minutes to understand, pity the next reader ;)